metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/usage.py

```diff
@@ -328,6 +332,22 @@ def usage_statistics(self, top_n: int) -> Iterator[UsageStatistic]:
                 column_freq=json.loads(row["column_freq"] or "[]"),
             )
 
+    def delete_original_read_events_for_view_query_events(self) -> None:
+        # CAUTION: Be careful when updating this query
+        # Ideally we should clean_active_object_cache python cache here
+        self.read_events.sql_query(
+            """
+            DELETE FROM
+                read_events
+            WHERE
+                read_events.from_query = 0 AND
+                read_events.name in (
+                    SELECT q.key FROM query_events q WHERE q.on_view = 1
+                )
+        """,
+            refs=[self.read_events, self.query_events],
```

Comment:
You can drop `self.read_events` from `refs` -- don't need to include itself

Reply:
removed